saleor/graphql/account/mutations/staff.py

```diff
@@ -0,0 +1,379 @@
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a staff user to delete.")
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        if not cls.check_permissions(info.context.user):
+            raise PermissionDenied()
+
+        user_id = data.get("id")
+        instance = cls.get_node_or_error(info, user_id, only_type=User)
+        cls.clean_instance(info, instance)
+
+        db_id = instance.id
+        remove_staff_member(instance)
+        # After the instance is deleted, set its ID to the original database's
+        # ID so that the success response contains ID of the deleted object.
+        instance.id = db_id
+        return cls.success_response(instance)
+
+
+class AddressCreate(ModelMutation):
+    user = graphene.Field(
+        User, description="A user instance for which the address was created."
+    )
+
+    class Arguments:
+        user_id = graphene.ID(
+            description="ID of a user to create address for", required=True
+        )
+        input = AddressInput(
+            description="Fields required to create address", required=True
+        )
+
+    class Meta:
+        description = "Creates user address"
+        model = models.Address
+        permissions = ("account.manage_users",)
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        user_id = data["user_id"]
+        user = cls.get_node_or_error(info, user_id, field="user_id", only_type=User)
+        response = super().perform_mutation(root, info, **data)
+        if not response.errors:
+            user.addresses.add(response.address)
+            response.user = user
+        return response
+
+
+class AddressUpdate(BaseAddressUpdate):
+    class Meta:
+        description = "Updates an address"
+        model = models.Address
+        permissions = ("account.manage_users",)
+
+
+class AddressDelete(BaseAddressDelete):
+    class Meta:
+        description = "Deletes an address"
+        model = models.Address
+        permissions = ("account.manage_users",)
+
+
+class AddressSetDefault(BaseMutation):
+    user = graphene.Field(User, description="An updated user instance.")
+
+    class Arguments:
+        address_id = graphene.ID(required=True, description="ID of the address.")
+        user_id = graphene.ID(
+            required=True, description="ID of the user to change the address for."
+        )
+        type = AddressTypeEnum(required=True, description="The type of address.")
+
+    class Meta:
+        description = "Sets a default address for the given user."
+        permissions = ("account.manage_users",)
+
+    @classmethod
+    def perform_mutation(cls, _root, info, address_id, user_id, **data):
+        address = cls.get_node_or_error(
+            info, address_id, field="address_id", only_type=Address
+        )
+        user = cls.get_node_or_error(info, user_id, field="user_id", only_type=User)
+
+        if address not in user.addresses.all():
```

Comment:
It loads all user's addresses to memory. I think that a better idea is to use `.exists()` or normal `filter()`

Reply:
Let's change this in all modified files